### ISSUE TITLE
EnvoyQuicClientSessionTests/EnvoyQuicClientSessionTest.EcnReporting is flaky

### ISSUE THREAD

**Issue by adisuissa** [envoyproxy/envoy#34582]
The test EnvoyQuicClientSessionTests/EnvoyQuicClientSessionTest.EcnReporting from /test/common/quic:envoy_quic_client_session_test is flaky.

```
[ RUN      ] EnvoyQuicClientSessionTests/EnvoyQuicClientSessionTest.EcnReporting/0
test/common/quic/envoy_quic_client_session_test.cc:570: Failure
Expected equality of these values:
  stats.num_ecn_marks_received.ect1
    Which is: 0
  1
Stack trace:
  0x55b075ed398e: Envoy::Quic::EnvoyQuicClientSessionTest_EcnReporting_Test::TestBody()
  0x55b07876be9f: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x55b0787669eb: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x55b07874ce56: testing::Test::Run()
  0x55b07874d879: testing::TestInfo::Run()
... Google Test internal frames ...

[external/com_google_absl/absl/flags/internal/flag.cc : 113] RAW: Restore saved value of envoy_reloadable_features_quic_receive_ecn to: true
[  FAILED  ] EnvoyQuicClientSessionTests/EnvoyQuicClientSessionTest.EcnReporting/0, where GetParam() = RFCv1 (112 ms)
```

[link to logs](https://dev.azure.com/cncf/envoy/_build/results?buildId=172275&view=logs&j=8bf29878-a4cc-50f7-4e84-2255e6fd4065&t=150a3762-112b-5713-0d28-af1eb6c2fbf4&l=3754).

cc @danzh2010

**Comment by danzh2010** [comment]
@martinduke

**Comment by martinduke** [comment]
I'm having a lot of trouble reproducing this; will try again tomorrow.

**Comment by martinduke** [comment]
Was reproducible when running 1000 tests, only when running the whole EnvoyQuicClientSessionTest rather than filtering for the EcnReporting question.

Adding more stats, the issue is that there are no packets arriving at all, not that they are arriving unmarked. There is some interaction between the other tests in the suite and this one; the easiest fix is to only check the  ECN marks if any packets have arrived.

**Comment by martinduke** [comment]
This was fixed by #34656
